orquesta/tests/unit/conducting/test_workflow_conductor_rerun.py

```diff
@@ -0,0 +1,630 @@
+        expected_term_ctx = {'loop': False}
+        self.assertDictEqual(conductor.get_workflow_terminal_context(), expected_term_ctx)
+        self.assertEqual(conductor.get_workflow_status(), statuses.SUCCEEDED)
+
+        # Check the list of terminal tasks.
+        term_tasks = conductor.workflow_state.get_terminal_tasks()
+        actual_term_tasks = [t['id'] for t in term_tasks]
+        expected_term_tasks = [task_name, task_name]
+        self.assertListEqual(actual_term_tasks, expected_term_tasks)
+
+    def test_workflow_rerun_one_task_with_multiple_tasks_failure(self):
+        # 1. Prepare rerun workflow
+        conductor = self.pre_rerun_workflow(PARALLEL_WF_DEF)
+
+        # 2. Fail task1 and task3
+        # Conduct tasks1 and task3.
+        self.forward_task_statuses(conductor, 'task1', [statuses.RUNNING, statuses.FAILED])
+        self.forward_task_statuses(conductor, 'task3', [statuses.RUNNING, statuses.FAILED])
+
+        # Check the list of terminal tasks.
+        term_tasks = conductor.workflow_state.get_terminal_tasks()
+        actual_term_tasks = [t['id'] for t in term_tasks]
+        expected_term_tasks = ['task1', 'task3']
+        self.assertListEqual(actual_term_tasks, expected_term_tasks)
+        self.assertEqual(conductor.get_workflow_status(), statuses.FAILED)
+        conductor.reset_workflow_output()
+
+        # Log the errors
+        errors = ['error1', 'error2']
+        conductor.log_errors(errors, 'task1')
+        conductor.log_errors(errors, 'task3')
+
+        # 3. Start rerun workflow only for task1
+        rerun_tasks = ['task1']
+        number_tasks = 2
+        self.assert_workflow_rerun_request(conductor, rerun_tasks, number_tasks)
+
+        # Conduct task1 and check context.
+        self.forward_task_statuses(conductor, 'task1', [statuses.RUNNING, statuses.SUCCEEDED])
+
+        expected_t1_ctx = {'var1': 'xyz', 'var2': 234}
+        expected_txsn_ctx = {'task2__t0': expected_t1_ctx}
+        self.assertDictEqual(conductor.get_task_transition_contexts('task1', 0), expected_txsn_ctx)
+
+        task_ids = ['task2']
+        task_ctxs = [expected_t1_ctx]
+        task_routes = [0]
+        self.assert_next_tasks(conductor, task_ids, task_ctxs, task_routes)
+
+        # Conduct task2 and check context.
+        self.forward_task_statuses(conductor, 'task2', [statuses.RUNNING, statuses.SUCCEEDED])
+        self.assertDictEqual(conductor.get_task_transition_contexts('task2', 0), {})
+        self.assert_next_task(conductor, has_next_task=False)
+
+        # Check the list of terminal tasks.
+        term_tasks = conductor.workflow_state.get_terminal_tasks()
+        actual_term_tasks = [t['id'] for t in term_tasks]
+        expected_term_tasks = ['task1', 'task3', 'task2']
+        self.assertListEqual(actual_term_tasks, expected_term_tasks)
+
+        # Check workflow status, context and output.
+        self.assertEqual(conductor.get_workflow_status(), statuses.SUCCEEDED)
+
```

Comment:
Can you add another unit test similar to `test_workflow_rerun_one_task_with_multiple_tasks_failure` but rerun task1 and task3 separately?